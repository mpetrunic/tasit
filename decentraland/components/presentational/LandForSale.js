import React from "react";
import { StyleSheet, View, Text } from "react-native";
import PropTypes from "prop-types";
import { responsiveWidth } from "react-native-responsive-dimensions";
import Estate from "./Estate";

export default function LandForSale({ landForSale }) {
  const { asset: estate, priceMana } = landForSale;

  // Note: Conversion to USD will be implemented on v0.2.0
  // <Text>{priceMana} MANA (~${landForSale.priceUSD})</Text>
  return (
    <View style={styles.landContainer}>
      <Estate estate={estate} />
      <Text>{priceMana} MANA</Text>
    </View>
  );
}

LandForSale.propTypes = {
  landForSale: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  landContainer: {
    width: responsiveWidth(60),
  },
});
