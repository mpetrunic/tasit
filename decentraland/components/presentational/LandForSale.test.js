import React from "react";
import { shallow } from "enzyme";
import LandForSale from "./LandForSale";

describe("LandForSale", () => {
  it("renders the component", async () => {
    const landForSale = {
      id: -1,
      priceMana: 0,
      priceUSD: 0,
      asset: {
        id: -1,
        name: "Not found",
        img: "https://decentraland.org/images/logo-65f7b27caf.png",
      },
    };
    expect(
      shallow(<LandForSale landForSale={landForSale} />)
    ).toMatchSnapshot();
  });
});
