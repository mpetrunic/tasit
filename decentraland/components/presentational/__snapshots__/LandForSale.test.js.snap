// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandForSale renders the component 1`] = `
<Component
  style={
    Object {
      "width": 712.5,
    }
  }
>
  <Estate
    estate={
      Object {
        "id": "123",
        "img": "https://api.decentraland.org/v1/estates/1/map.png",
        "name": "Sample Estate",
        "type": "ESTATE",
      }
    }
  />
  <Component>
    0
     MANA
  </Component>
</Component>
`;
